spring-rabbit/src/main/java/org/springframework/amqp/rabbit/core/RabbitTemplate.java

```diff
@@ -1750,6 +1766,59 @@ public Boolean isMandatoryFor(final Message message) {
 			throw convertRabbitAccessException(ex);
 		}
 		finally {
+			if (!invokeScope) {
+				if (resourceHolder != null) {
+					ConnectionFactoryUtils.releaseResources(resourceHolder);
+				}
+				else {
+					RabbitUtils.closeChannel(channel);
+					RabbitUtils.closeConnection(connection);
+				}
+			}
+		}
+	}
+
+	@Override
+	public <T> T invoke(TemplateCallback<T> action) {
+		final Channel currentChannel = this.dedicatedChannels.get();
+		Assert.state(currentChannel == null, () -> "Nested invoke() calls are not supported; channel '" + currentChannel
+				+ "' is already associated with this thread");
+		this.activeTemplateCallbacks.incrementAndGet();
+		RabbitResourceHolder resourceHolder = null;
+		Connection connection = null; // NOSONAR (close)
+		Channel channel;
+		if (isChannelTransacted()) {
+			resourceHolder = ConnectionFactoryUtils.getTransactionalResourceHolder(getConnectionFactory(), true);
```

Comment:
Doesn't this mean that we reuse `channel` in case of outer transaction? But with the *dedicated* channel like this PR claims...
Looks like  `ConnectionFactoryUtils.getTransactionalResourceHolder()` returns exactly:
```
RabbitResourceHolder resourceHolder = (RabbitResourceHolder) TransactionSynchronizationManager
				.getResource(connectionFactory);
```